From 6f9ecd6f2607b58fb5f30572ca47d02da1f0c743 Mon Sep 17 00:00:00 2001
From: Darren Kenny <darren.kenny@oracle.com>
Date: Tue, 24 Nov 2020 18:04:22 +0000
Subject: gnulib/regcomp: Fix uninitialized re_token

This issue has been fixed in the latest version of gnulib, so to
maintain consistency, I've backported that change rather than doing
something different.

Fixes: CID 73828

Signed-off-by: Darren Kenny <darren.kenny@oracle.com>
Reviewed-by: Daniel Kiper <daniel.kiper@oracle.com>

Patch-Name: 2021-02-security/025-gnulib-regcomp-Fix-uninitialized-re_token.patch
---
 conf/Makefile.extra-dist                       |  1 +
 grub-core/gnulib-fix-regcomp-uninit-token.diff | 12 ++++++++++++
 grub-core/gnulib/regcomp.c                     |  3 +--
 3 files changed, 14 insertions(+), 2 deletions(-)
 create mode 100644 grub-core/gnulib-fix-regcomp-uninit-token.diff

diff --git a/conf/Makefile.extra-dist b/conf/Makefile.extra-dist
index a22f7f016..9ab5a681e 100644
--- a/conf/Makefile.extra-dist
+++ b/conf/Makefile.extra-dist
@@ -30,6 +30,7 @@ EXTRA_DIST += grub-core/genemuinitheader.sh
 
 EXTRA_DIST += grub-core/gnulib-fix-null-deref.diff
 EXTRA_DIST += grub-core/gnulib-fix-null-state-deref.diff
+EXTRA_DIST += grub-core/gnulib-fix-regcomp-uninit-token.diff
 EXTRA_DIST += grub-core/gnulib-fix-regexec-null-deref.diff
 EXTRA_DIST += grub-core/gnulib-fix-uninit-structure.diff
 EXTRA_DIST += grub-core/gnulib-fix-unused-value.diff
diff --git a/grub-core/gnulib-fix-regcomp-uninit-token.diff b/grub-core/gnulib-fix-regcomp-uninit-token.diff
new file mode 100644
index 000000000..24d96990e
--- /dev/null
+++ b/grub-core/gnulib-fix-regcomp-uninit-token.diff
@@ -0,0 +1,12 @@
+--- a/lib/regcomp.c	2020-11-24 17:06:08.159223858 +0000
++++ b/lib/regcomp.c	2020-11-24 17:06:15.630253923 +0000
+@@ -3790,8 +3790,7 @@
+ create_tree (re_dfa_t *dfa, bin_tree_t *left, bin_tree_t *right,
+ 	     re_token_type_t type)
+ {
+-  re_token_t t;
+-  t.type = type;
++  re_token_t t = { .type = type };
+   return create_token_tree (dfa, left, right, &t);
+ }
+ 
diff --git a/grub-core/gnulib/regcomp.c b/grub-core/gnulib/regcomp.c
index de9f62208..6d0830ac6 100644
--- a/grub-core/gnulib/regcomp.c
+++ b/grub-core/gnulib/regcomp.c
@@ -3790,8 +3790,7 @@ static bin_tree_t *
 create_tree (re_dfa_t *dfa, bin_tree_t *left, bin_tree_t *right,
 	     re_token_type_t type)
 {
-  re_token_t t;
-  t.type = type;
+  re_token_t t = { .type = type };
   return create_token_tree (dfa, left, right, &t);
 }
 
